engine/test-utils/src/main/java/io/deephaven/engine/testutil/assertj/ColumnSourceValuesAssert.java

```diff
@@ -0,0 +1,192 @@
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.rowset.RowSet.Iterator;
+import io.deephaven.engine.table.ChunkSource.FillContext;
+import io.deephaven.engine.table.ColumnSource;
+import org.assertj.core.api.ByteArrayAssert;
+import org.assertj.core.api.CharArrayAssert;
+import org.assertj.core.api.DoubleArrayAssert;
+import org.assertj.core.api.FloatArrayAssert;
+import org.assertj.core.api.IntArrayAssert;
+import org.assertj.core.api.IteratorAssert;
+import org.assertj.core.api.LongArrayAssert;
+import org.assertj.core.api.ShortArrayAssert;
+
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ColumnSourceValuesAssert<T> {
+
+    private final ColumnSource<T> actual;
+    private final RowSet rowSet;
+
+    public ColumnSourceValuesAssert(ColumnSource<T> columnSource, RowSet rowSet) {
+        this.actual = Objects.requireNonNull(columnSource);
+        this.rowSet = Objects.requireNonNull(rowSet);
+    }
+
+    public CharArrayAssert chars() {
+        return chars(false);
+    }
+
+    public ByteArrayAssert bytes() {
+        return bytes(false);
+    }
+
+    public ShortArrayAssert shorts() {
+        return shorts(false);
+    }
+
+    public IntArrayAssert ints() {
+        return ints(false);
+    }
+
+    public LongArrayAssert longs() {
+        return longs(false);
+    }
+
+    public FloatArrayAssert floats() {
+        return floats(false);
+    }
+
+    public DoubleArrayAssert doubles() {
+        return doubles(false);
+    }
+
+    public CharArrayAssert chars(boolean prev) {
+        return assertThatChars(actual.cast(char.class), rowSet, prev);
+    }
+
+    public ByteArrayAssert bytes(boolean prev) {
+        return assertThatBytes(actual.cast(byte.class), rowSet, prev);
+    }
+
+    public ShortArrayAssert shorts(boolean prev) {
+        return assertThatShorts(actual.cast(short.class), rowSet, prev);
+    }
+
+    public IntArrayAssert ints(boolean prev) {
+        return assertThatInts(actual.cast(int.class), rowSet, prev);
+    }
+
+    public LongArrayAssert longs(boolean prev) {
+        return assertThatLongs(actual.cast(long.class), rowSet, prev);
+    }
+
+    public FloatArrayAssert floats(boolean prev) {
+        return assertThatFloats(actual.cast(float.class), rowSet, prev);
+    }
+
+    public DoubleArrayAssert doubles(boolean prev) {
+        return assertThatDoubles(actual.cast(double.class), rowSet, prev);
+    }
+
+    public static CharArrayAssert assertThatChars(ColumnSource<Character> source, RowSet rowSet, boolean prev) {
+        final char[] out = new char[rowSet.intSize()];
+        try (final Iterator it = rowSet.iterator()) {
+            final IteratorAssert<Long> itAssert = assertThat(it);
+            for (int i = 0; i < out.length; ++i) {
+                itAssert.hasNext();
+                out[i] = prev ? source.getPrevChar(it.nextLong()) : source.getChar(it.nextLong());
```

Comment:
So in the other cases we are using chunks, but not this case.  What is the reasoning?